### ISSUE TITLE
GdkPixbuf loader (as used by eog) breaks on large files

### ISSUE THREAD

**Issue by 190n** [libjxl/libjxl#1232]
**Describe the bug**

I can't open certain large JPEG XL files (the breaking point is somewhere between 4097x2731 and 6000x4000; I can do a more exhaustive test if needed) in eog. This holds for lossless and lossy files, as well as losslessly transcoded JPEGs. I'm pretty sure that these files used to work in eog, although I'm not certain. I don't think this is a bug in GdkPixbuf or eog, since the full-resolution source JPEGs (6000x4000) open just fine.

**To Reproduce**

- Try to open a large JPEG XL image (one that breaks is attached) with eog

**Expected behavior**

The image is displayed.

**Screenshots**
If applicable, add screenshots or example input/output images to help explain your problem.

**Environment**
 - OS: Arch Linux
 - Compiler version: clang 13.0.0 (what Arch's package was built with)
 - CPU type: x86_64
 - cjxl/djxl version string: cjxl v0.6.1 a205468b [AVX2,SSE4,SSSE3,Scalar]
 - GdkPixbuf version: 2.42.6-2
 - eog version: 41.1

**Additional context**

Log output from eog:

```

(eog:264174): GdkPixbuf-CRITICAL **: 23:41:50.805: gdk_pixbuf_animation_is_static_image: assertion 'GDK_IS_PIXBUF_ANIMATION (animation)' failed

(eog:264174): GdkPixbuf-CRITICAL **: 23:41:50.805: gdk_pixbuf_animation_get_iter: assertion 'GDK_IS_PIXBUF_ANIMATION (animation)' failed

(eog:264174): GdkPixbuf-CRITICAL **: 23:41:50.805: gdk_pixbuf_animation_iter_get_pixbuf: assertion 'GDK_IS_PIXBUF_ANIMATION_ITER (iter)' failed

(eog:264174): EOG-WARNING **: 23:41:50.902: Thumbnail creation failed

(eog:264174): EOG-WARNING **: 23:41:50.902: Thumbnail creation failed

(eog:264174): EOG-WARNING **: 23:41:50.902: Thumbnail creation failed

(eog:264174): EOG-WARNING **: 23:41:50.902: Thumbnail creation failed

```

ZIP archive containing a file that breaks in eog (works with Chromium and djxl):

[broken-file.zip](https://github.com/libjxl/libjxl/files/8237276/broken-file.zip)

**Comment by mo271** [comment]
Thanks for the report!

I tried to reproduce the issue and have pretty similar settings/versions, except that I used current main fbcd01984bb6d724eaf3bf07955269787ad60cc7. There it seems to work and I can view the image `big.jxl` you provide just fine in eog.  
Could you try building with the current main and see if you still have the issue?

**Comment by 190n** [comment]
Oh yeah, seems like it's been fixed since 0.6.1. I'll run the git build for now and see if I can figure out where it got fixed with `git bisect`.
